src/Razor/src/Microsoft.VisualStudio.Mac.LanguageServices.Razor/Documents/RazorDocumentControllerExtension.cs

```diff
@@ -0,0 +1,47 @@
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.Text;
+using MonoDevelop.Core;
+using MonoDevelop.Ide.Gui.Documents;
+
+namespace Microsoft.VisualStudio.Editor.Razor.Documents
+{
+    /// <summary>
+    /// This is a way to notify the VisualStudioMacEditorDocumentManager when Razor documents are
+    /// opened or closed.
+    /// TODO: add support for an open document being renamed, when the MonoDevelop API is available.

```

Comment:
@KirillOsenkov is this in the process of being done? Also, if a user does rename the doc what happens?

Reply:
We have an issue tracking this, but we won't likely get to it soon. For now it's OK to ship with this.